operator/controller/dnsnameresolver/controller.go

```diff
@@ -129,7 +127,7 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		// Check if the DNSNameResolver resource is deleted. If so, delete DNS names matching the DNSNameResolver resource
 		// from the resolver.
 		if errors.IsNotFound(err) {
-			r.resolver.Delete(request.Name)
+			r.resolver.DeleteDNSNameResolver(request.Name)
```

Comment:
Can this be consistent in naming with the Add method? Something like
```suggestion
			r.resolver.DeleteResolvecName(request.Name)
```